content/en/docs/ops/common-problems/security-issues/index.md

```diff
@@ -79,42 +79,62 @@ With Istio, you can enable authentication for end users through [request authent
 
-1. Make sure that your authorization policies are in the right namespace (as specified in `metadata/namespace` field).
+To fix the problem, just remove the extra `-` to make the policy have only 1 rule that allows request if the
+path is `/foo` **and** the source namespace is `foo`, which is more restrictive.
 
-1. Make sure that your authorization policies with ALLOW action don't use any HTTP only fields for TCP traffic.
-Otherwise, Istio ignores the ALLOW policies as if they don't exist.
+### Make sure you are NOT using HTTP-only fields on TCP ports
 
-1. An HTTP response with the value `RBAC: Access Denied` indicates an authorization policy is in effect. You can determine the authorization policy in effect by running `istioctl x authz check POD-NAME.POD-NAMESPACE`.
+The authorization policy will be more restrictive because HTTP-only fields (e.g. `host`, `path`, `headers`, JWT, etc.)
+will never be matched on raw TCP connections. This could cause unexpected 403 denies.
 
-1. Make sure that your authorization policies with DENY action don't use any HTTP only fields for TCP traffic.
-Otherwise, Istio ignores the rules with HTTP only fields within the DENY policies as if they don't exist.
+Check the Kubernetes service definition to verify that the port is [named with the correct protocol properly](/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection).
+If you are using HTTP-only fields on the port, make sure it has the `http-` prefix.
 
-1. An HTTP response with the value `upstream connect error or disconnect/reset before headers. reset reason: connection termination` can indicate an authorization policy with HTTP only fields applied to TCP traffic. Read the [port selection documentation](/docs/ops/configuration/traffic-management/protocol-selection/) for how Istio determines whether a service is using the http or tcp protocol.
+### Make sure the policy is applied to the correct target
 
-## Authorization is too permissive
+Check the workload selector and namespace to confirm it's applied to the right targets. You can determine the
+authorization policy in effect by running `istioctl x authz check POD-NAME.POD-NAMESPACE`.
 
-If authorization checks are enabled for a service and yet requests to the
-service aren't being blocked, then authorization was likely not enabled
-successfully. To verify, follow these steps:
+### Pay attention to the action specified in the policy
 
-1. Check the [authorization concept documentation](/docs/concepts/security/#authorization)
-   to correctly apply Istio authorization.
+- If not specified, the policy defaults to use action `ALLOW`.
 
-1. Make sure there is no typo in your policy YAML file. Especially check to make sure the authorization policy is applied
-to the right workload and namespace.
+- When `CUSTOM`, `ALLOW` and/or `DENY` are used together, all of them must be satisfied to allow a request. In other words,
```

Comment:
nit: may help link to the doc that explains how these 3 interact?

Reply:
Actually I think I will update the `concept` page in a follow up PR and then add the link, currently this is explained in the design doc.